src/Sylius/Bundle/AddressingBundle/Repository/ZoneRepository.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\AddressingBundle\Repository;
+
+use Doctrine\ORM\QueryBuilder;
+use Sylius\Bundle\ResourceBundle\Doctrine\ORM\EntityRepository;
+use Sylius\Component\Addressing\Model\AddressInterface;
+use Sylius\Component\Addressing\Model\Scope;
+use Sylius\Component\Addressing\Model\ZoneInterface;
+use Sylius\Component\Addressing\Repository\ZoneRepositoryInterface;
+
+/**
+ * @implements ZoneRepositoryInterface<ZoneInterface>
+ */
+class ZoneRepository extends EntityRepository implements ZoneRepositoryInterface
+{
+    public function findOneByAddress(AddressInterface $address, ?string $scope = null): ?ZoneInterface
+    {
+        $query = $this->createFindByAddressQueryBuilder($address, $scope);
+
+        $query
+            ->addSelect('(CASE
+                    WHEN z.type = \'province\' THEN 1
```

Comment:
Can those values be converted to const usage? (`ZoneInterface::TYPE_PROVIENCE`, etc)